src/queue/index.ts

```diff
@@ -0,0 +1,10 @@
+import type { ComposeRequest } from '@app/api/composes';
+import type { Worker } from '@app/worker';
+
+import { JobQueue } from './queue';
+
+export const createQueue = (worker: Worker<ComposeRequest>) => {
```

Comment:
So we could use something like https://bullmq.io/ here instead of re-inventing the wheel.
The downside is that we would need to have a redis server running since it relies on that.

So it's a bit of a trade-off